### PR TITLE
Add MangaEsp

### LINKED ISSUE #479: Source Request: MangaEsp
### Source name

MangaEsp

### Source link

https://mangaesp.co/

### Source language

Español

### Other details

_No response_

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/keiyoushi/extensions-source/) and verified it does not appear in the code base.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by bapeey**
Closes #479 

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/es/mangaesp/src/eu/kanade/tachiyomi/extension/es/mangaesp/MangaEsp.kt [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                        "created_at" -> filteredList.sortByDescending { it.createdAt }
+                    }
+                }
+            }
+        }
+
+        val hasNextPage = filteredList.size > page * MANGAS_PER_PAGE
+
+        return MangasPage(
+            filteredList.subList((page - 1) * MANGAS_PER_PAGE, filteredList.size)
```
Comment (AwkwardPeak7) [inline src/es/mangaesp/src/eu/kanade/tachiyomi/extension/es/mangaesp/MangaEsp.kt]: wouldn't this just return the original list back? probably need to do something like:
 
```suggestion
            filteredList.subList((page - 1) * MANGAS_PER_PAGE, min(page * MANGAS_PER_PAGE, filteredList.size)
```
don't forget to import `kotlin.math.min`

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: src/es/mangaesp/src/eu/kanade/tachiyomi/extension/es/mangaesp/MangaEsp.kt [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+
+class MangaEsp : HttpSource() {
+
+    override val name = "MangaEsp"
+
+    override val baseUrl = "https://mangaesp.co"
+
+    private val apiBaseUrl = "https://apis.mangaesp.co"
+
+    private val imgCdnUrl = "https://cdn.statically.io/img"
```
Comment (stevenyomi) [inline src/es/mangaesp/src/eu/kanade/tachiyomi/extension/es/mangaesp/MangaEsp.kt]: Did you check if Statically work on their site? Afaik they redirect image requests to the original URL for a lot of sites.

Comment (bapeey) [reply src/es/mangaesp/src/eu/kanade/tachiyomi/extension/es/mangaesp/MangaEsp.kt]: Ah, its true. I'm just replicating what the site does but I think I can skip the redirect

File: src/es/mangaesp/src/eu/kanade/tachiyomi/extension/es/mangaesp/MangaEsp.kt (hunk 1 above)
Comment (bapeey) [reply src/es/mangaesp/src/eu/kanade/tachiyomi/extension/es/mangaesp/MangaEsp.kt]: true, my bad

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**